ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml

```diff
@@ -0,0 +1,161 @@
+  tags: [jenkins_user, adoptopenjdk]
+
+- name: Check if pkgutil is installed
+  shell: pkgutil --version
+  ignore_errors: true
+  register: pkgutil_installed
+  tags:
+    - build_tools
+
+- name: Install pkgutil
+  shell: |
+    set -o pipefail
+    echo action=nocheck > responses.txt
+    yes all | pkgadd -d http://get.opencsw.org/now -a responses.txt
+    /opt/csw/bin/pkgutil -U
+    /opt/csw/bin/pkgutil -y -i pkgutil
+    /usr/sbin/pkgchk -L CSWpkgutil
+  when: pkgutil_installed.rc != 0
+  tags: build_tools
+
+- name: Install Build Tool Packages
+  pkgutil: "name={{ item }} state=present"
+  with_items: "{{ Build_Tool_Packages }}"
+  tags: build_tools
+
+- name: Check if Solaris Studio 12.3 is installed
+  stat:
+    path: /opt/solarisstudio12.3/bin/
+  register: solaris_studio
+  tags:
+    - build_tools
+    - solaris_studio
+    - adoptopenjdk
+
+- name: Check if Solaris Studio 12.3 is downloaded
+  stat:
+    path: /tmp/SolarisStudio12.3-solaris-x86-pkg.tar.bz2
+  register: solaris_studio_downloaded
+  when: not solaris_studio.stat.exists
+  tags:
+    - build_tools
+    - solaris_studio
+    - adoptopenjdk
+
+# Stored in Azure Blob Storage (SAS URL set to expire in 2030)
+- name: Download Solaris Studio 12.3
+  # Original file is at https://www.oracle.com/tools/developerstudio/downloads/solaris-studio-v123-downloads.html
+  command: wget "https://ansiblestorageadopt.blob.core.windows.net/solarisstudiox64/SolarisStudio12.3-solaris-x86-pkg.tar.bz2?{{ SolarisStudio12_solaris_x86_SAS_TOKEN }}" -O /tmp/SolarisStudio12.3-solaris-x86-pkg.tar.bz2
+  when: not solaris_studio.stat.exists and not solaris_studio_downloaded.stat.exists
+  tags:
+    - build_tools
+    - solaris_studio
+    - adoptopenjdk
+
+- name: Extract Solaris Studio 12.3
+  shell: |
+    set -o pipefail
+    bzcat SolarisStudio12.3-solaris-x86-pkg.tar.bz2 | /bin/tar -xf -
+  args:
+    chdir: /tmp
+  tags:
+    - build_tools
+    - solaris_studio
+    - adoptopenjdk
+  when: not solaris_studio.stat.exists
+
+- name: Install Solaris Studio 12.3
+  shell: ./SolarisStudio12.3-solaris-x86-pkg/solarisstudio.sh --non-interactive
+  args:
+    chdir: /tmp
+  tags:
+    - build_tools
+    - solaris_studio
+    - adoptopenjdk
+  when: not solaris_studio.stat.exists
+
+- name: Check if cups is installed
+  stat:
+    path: /opt/sfw/cups
+  register: cups_installed
+  tags:
+    - build_tools
+    - cups
+
+- name: Download cups 1.4.4
+  command: wget https://github.com/apple/cups/releases/download/release-1.4.4/cups-1.4.4-source.tar.gz -O /tmp/cups-1.4.4-source.tar.gz
+  when: not cups_installed.stat.exists
+  tags:
+    - build_tools
+    - cups
+
+- name: Extract cups 1.4.4
+  shell: tar -xf cups-1.4.4-source.tar.gz
+  args:
+    chdir: /tmp
+  when: not cups_installed.stat.exists
+  tags:
+    - build_tools
+    - cups
+
+- name: Compile cups 1.4.4
+  shell: |
+    cd /tmp/cups-1.4.4
+    ./configure --disable-gssapi --disable-ldap --prefix=/opt/sfw/cups
```

Comment:
Since the version we want isn't in the sfw/csw repos would it not be preferable to put this into `/usr/local/cups` instead of other namespaces?

Reply:
This is the place that OpenJDK expects cups to be located. It's also where we put cups on the other solaris servers